File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/PlusNotificationForm.kt
package com.sdkwork.backend

data class PlusNotificationForm(
    val role: String? = null,
    val senderId: Int? = null,
    val sender: Participant? = null,
    val receiverId: Int? = null,
    val receiver: Participant? = null,
    val groupId: Int? = null,
    val title: String? = null,
    val content: String? = null,
    val type: String? = null,
    val channelType: String? = null,
    val templateId: String? = null,
    val templateParams: Map<String, Any>? = null,
    val redirectUrl: String? = null,
    val miniProgramPath: String? = null,
    val status: String? = null,
    val sentAt: String? = null,
    val readAt: String? = null,
    val extraData: Map<String, Any>? = null,
    val retryCount: Int? = null,
    val maxRetryCount: Int? = null,
    val errorMessage: String? = null
)

File: sdkwork-backend-sdk-typescript/src/types/plus-notification-form.ts
import type { Participant } from './participant';

/** Notification creation form */
export interface PlusNotificationForm {
  /** Message role */
  role: 'DEFAULT' | 'USER' | 'ASSISTANT' | 'SYSTEM' | 'FUNCTION';
  /** Sender ID */
  senderId?: number;
  /** Sender */
  sender?: Participant;
  /** Receiver ID */
  receiverId?: number;
  /** Receiver */
  receiver?: Participant;
  /** Group ID */
  groupId?: number;
  /** Title */
  title?: string;
  /** Content */
  content?: string;
  /** Type */
  type: 'USER' | 'GROUP' | 'SYSTEM' | 'TASK' | 'MESSAGE' | 'ALERT' | 'USER_SYSTEM' | 'USER_GROUP' | 'GROUP_SYSTEM' | 'OTHER';
  /** Channel type */
  channelType: 'DEFAULT' | 'EMAIL' | 'SMS' | 'IN_APP' | 'WECHAT' | 'WECHAT_TEMPLATE' | 'WECHAT_MINI_PROGRAM' | 'DINGTALK' | 'IM' | 'PUSH' | 'OTHER';
  /** Template ID */
  templateId?: string;
  /** Template params */
  templateParams?: Record<string, Record<string, unknown>>;
  /** Redirect URL */
  redirectUrl?: string;
  /** Mini program path */
  miniProgramPath?: string;
  /** Status */
  status: 'PENDING' | 'SENDING' | 'SENT' | 'DELIVERED' | 'READ' | 'FAILED' | 'REVOKED' | 'EXPIRED';
  /** Sent at */
  sentAt?: string;
  /** Read at */
  readAt?: string;
  /** Extra data */
  extraData?: Record<string, Record<string, unknown>>;
  /** Retry count */
  retryCount?: number;
  /** Max retry count */
  maxRetryCount?: number;
  /** Error message */
  errorMessage?: string;
}
